sound/soc/sof/intel/hda.c

```diff
@@ -754,14 +754,6 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	sdev->pdata->hw_pdata = hdev;
 	hdev->desc = chip;
 
-	hdev->dmic_dev = platform_device_register_data(sdev->dev, "dmic-codec",
-						       PLATFORM_DEVID_NONE,
-						       NULL, 0);
-	if (IS_ERR(hdev->dmic_dev)) {
-		dev_err(sdev->dev, "error: failed to create DMIC device\n");
-		return PTR_ERR(hdev->dmic_dev);
-	}
-
```

Comment:
do we want to move the dmic device creation to the client?
This would force the client driver to check what platform it runs on, when this is not needed when you create the dmic device from a platform-specific probe.
If we want a generic client then we should keep this here?